ml-kem/src/kem.rs

```diff
@@ -26,6 +29,20 @@ where
     z: B32,
 }
 
+#[cfg(feature = "zeroize")]
+impl<P> Zeroize for DecapsulationKey<P>
+where
+    P: KemParams,
+{
+    fn zeroize(&mut self) {
+        self.dk_pke.zeroize();
+        self.z.zeroize();
+    }
+}
```

Comment:
I would suggest this just be a `Drop` impl:

```suggestion
#[cfg(feature = "zeroize")]
impl<P> Drop for DecapsulationKey<P>
where
    P: KemParams,
{
    fn drop(&mut self) {
        self.dk_pke.zeroize();
        self.z.zeroize();
    }
}
```

I'm also a bit taken aback that `ZeroizeOnDrop` doesn't actually bound on `Drop`, which seems like a bug. I guess we should probably fix that although it'd be a breaking change.

I'd only recommend impl'ing `Zeroize` on lower-level types that only potentially contain secrets, because there's a possible "use after zeroize" condition possible. Types like keys that definitely contain secrets can always take care of cleaning up those secrets in a `Drop` impl, and `ZeroizeOnDrop` is specifically supposed to be a marker trait that means that such a `Drop` impl is in place.